engine/src/configuration/extended_conf.cc

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2024 Centreon
+ *
+ * This file is part of Centreon Engine.
+ *
+ * Centreon Engine is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 2
+ * as published by the Free Software Foundation.
+ *
+ * Centreon Engine is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Centreon Engine. If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include "com/centreon/engine/configuration/extended_conf.hh"
+#include "com/centreon/engine/configuration/state.hh"
+#include "com/centreon/exceptions/msg_fmt.hh"
+
+using namespace com::centreon::engine::configuration;
+
+std::list<std::unique_ptr<extended_conf>> extended_conf::_confs;
+
+/**
+ * @brief Construct a new extended state::extended state object
+ *
+ * @param path
+ * @throw exception if json mal formed
+ */
+extended_conf::extended_conf(const std::string& path) : _path(path) {
+  if (::stat(_path.c_str(), &_file_info)) {
+    SPDLOG_LOGGER_ERROR(log_v2::config(), "can't access to {}", _path);
+    throw exceptions::msg_fmt("can't access to {}", _path);
+  }
+  try {
+    _content = common::rapidjson_helper::read_from_file(_path);
+    SPDLOG_LOGGER_INFO(log_v2::config(), "extended conf file {} loaded", _path);
+  } catch (const std::exception& e) {
+    SPDLOG_LOGGER_ERROR(
+        log_v2::config(),
+        "extended_conf::extended_conf : fail to read json content from {}: {}",
+        _path, e.what());
+    throw;
+  }
+}
+
+void extended_conf::reload() {
```

Comment:
An explanation about this function would be interesting. Why would we need to reload() the file and not just load() it ?